types/types.spec.ts

```diff
@@ -0,0 +1,297 @@
+import crypto, {
```

Comment:
This file is not _required_ unless we want linting/testing capabilities _within_ this repo. It was copied from the various *.spec.js` files within the repo. It is useful for making sure that the types match the usage of the module. However, it isn't ideal to have two versions of the 'tests' like this... At the moment, there isn't much of a nicer way to handle this that I'm aware of?